book/src/development/add_documentation.md

```diff
@@ -0,0 +1,46 @@
+# Adding Documentation for New Lint
```

Comment:
There shouldn't be any story telling in here. Just the code snippet how the declaration/documentation looks like and a short explanation for each section in the documentation.